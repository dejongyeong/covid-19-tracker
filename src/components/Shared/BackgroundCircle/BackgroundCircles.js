import styled from "styled-components";

export const Circle1 = styled.div`
  height: 747px;
  width: 747px;
  position: absolute;
  top: -300px;
  right: -250px;
  background-color: ${(props) => props.theme.colors.secondaryLight};
  opacity: 50%;
  transform: rotate(-46.28deg);
  border-radius: 50%;
  -moz-border-radius: 50%;
  -webkit-border-radius: 50%;
  z-index: -1;

  @media ${(props) => props.theme.breakpoints.sm} {
    height: 482.21px;
    width: 482.21px;
    transform: rotate(-63.78deg);
    top: -241.11px;
    right: -241.11px;
  }
`;

export const Circle2 = styled.div`
  height: 297.56px;
  width: 297.56px;
  position: absolute;
  bottom: -148.78px;
  left: -100px;
  background-color: ${(props) => props.theme.colors.secondaryLight};
  opacity: 50%;
  transform: rotate(-28.59deg);
  border-radius: 50%;
  -moz-border-radius: 50%;
  -webkit-border-radius: 50%;
  z-index: -1;

  @media ${(props) => props.theme.breakpoints.sm} {
    bottom: -130px;
    left: -120px;
  }
`;
