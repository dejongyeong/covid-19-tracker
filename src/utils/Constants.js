const SIZE = {
  xs: "0px",
  sm: "480px",
  md: "768px",
  lg: "992px",
  xl: "1200px",
};

/* eslint-disable */
export const MEDIA_QUERIES = {
  xs: `(min-width:${SIZE.xs})`,
  sm: `(max-width:calc(${SIZE.sm} - 0.02px))`,
  md: `(max-width:calc(${SIZE.md} - 0.02px))`,
  lg: `(max-width:calc(${SIZE.lg} - 0.02px))`,
  xl: `(min-width:${SIZE.xl})`,
};
