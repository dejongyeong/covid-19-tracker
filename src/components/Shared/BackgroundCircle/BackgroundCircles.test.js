import "jest-styled-components";
import renderer from "react-test-renderer";
import { render, cleanup } from "@testing-library/react";

import { Circle1, Circle2 } from "./BackgroundCircles";
import { MEDIA_QUERIES } from "../../../utils/Constants";

afterEach(cleanup);

test("Circle 1 Snapshot", () => {
  const { container } = render(<Circle1 />);
  expect(container.firstChild).toMatchSnapshot();
});

test("Circle 2 Snapshot", () => {
  const { container } = render(<Circle2 />);
  expect(container.firstChild).toMatchSnapshot();
});

test("Circle 1 Width and Height", () => {
  const tree = renderer.create(<Circle1 />).toJSON();
  expect(tree).toHaveStyleRule("width", "747px");
  expect(tree).toHaveStyleRule("width", "482.21px", {
    media: `${MEDIA_QUERIES.sm}`,
  });
  expect(tree).toHaveStyleRule("height", "747px");
  expect(tree).toHaveStyleRule("height", "482.21px", {
    media: `${MEDIA_QUERIES.sm}`,
  });
});

test("Circle 2 Width and Height", () => {
  const tree = renderer.create(<Circle2 />).toJSON();
  expect(tree).toHaveStyleRule("width", "297.56px");
  expect(tree).not.toHaveStyleRule("width", "297.56px", {
    media: `${MEDIA_QUERIES.sm}`,
  });
  expect(tree).toHaveStyleRule("height", "297.56px");
  expect(tree).not.toHaveStyleRule("height", "297.56px", {
    media: `${MEDIA_QUERIES.sm}`,
  });
});
