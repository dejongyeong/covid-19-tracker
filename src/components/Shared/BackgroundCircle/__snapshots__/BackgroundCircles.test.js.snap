// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle Snapshots renders circle 1 correctly 1`] = `
.c0 {
  height: 747px;
  width: 747px;
  position: absolute;
  top: -300px;
  right: -250px;
  background-color: #eae2b7;
  opacity: 50%;
  -webkit-transform: rotate(-46.28deg);
  -ms-transform: rotate(-46.28deg);
  transform: rotate(-46.28deg);
  border-radius: 50%;
  -moz-border-radius: 50%;
  -webkit-border-radius: 50%;
  z-index: -1;
}

@media (max-width:calc(480px - 0.02px)) {
  .c0 {
    height: 482.21px;
    width: 482.21px;
    -webkit-transform: rotate(-63.78deg);
    -ms-transform: rotate(-63.78deg);
    transform: rotate(-63.78deg);
    top: -241.11px;
    right: -241.11px;
  }
}

<div
  className="c0"
/>
`;

exports[`Circle Snapshots renders circle 2 correctly 1`] = `
.c0 {
  height: 297.56px;
  width: 297.56px;
  position: absolute;
  bottom: -148.78px;
  left: -100px;
  background-color: #eae2b7;
  opacity: 50%;
  -webkit-transform: rotate(-28.59deg);
  -ms-transform: rotate(-28.59deg);
  transform: rotate(-28.59deg);
  border-radius: 50%;
  -moz-border-radius: 50%;
  -webkit-border-radius: 50%;
  z-index: -1;
}

@media (max-width:calc(480px - 0.02px)) {
  .c0 {
    bottom: -130px;
    left: -120px;
  }
}

<div
  className="c0"
/>
`;
